### PR TITLE
feat: support unix-domain-socket

### PR DESCRIPTION
**Pull request by fishbar**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [x] commit message follows commit guidelines

##### Description of change

let client support all options which is support by build-in api `socket.connect(options)`

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -75,6 +77,8 @@ describe('test/index.test.js', () => {
   }
 
   let client;
+  let client2;
+  const sockPath = path.join(require('os').tmpdir(), 'tcp-base-test.sock');
```
Comment (fengmk2) [inline test/index.test.js]: `os.tmpdir()`

Comment (fishbar) [reply test/index.test.js]: fixed

**Comment by fishbar** [comment]
挂掉的那个并发测试case是不是不稳定？ @gxcsoccer

**Comment by fishbar** [comment]
@gxcsoccer  绿了，合，升？

**Comment by gxcsoccer** [comment]
3.1.0

**Pull request merged**